src/Microsoft.ML.AutoML/API/RecommendationExperiment.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.ML.Data;
+
+namespace Microsoft.ML.AutoML
+{
+    public sealed class RecommendationExperimentSettings : ExperimentSettings
+    {
+        public RegressionMetric OptimizingMetric { get; set; }
+
+        public ICollection<RecommendationTrainer> Trainers { get; }
+
+        public RecommendationExperimentSettings()
+        {
+            OptimizingMetric = RegressionMetric.RSquared;
+            Trainers = Enum.GetValues(typeof(RecommendationTrainer)).OfType<RecommendationTrainer>().ToList();
+        }
+    }
+
+    public enum RecommendationTrainer
+    {
+        MatrixFactorization
+    }
+
+    public sealed class RecommendationExperiment : ExperimentBase<RegressionMetrics, RecommendationExperimentSettings>
+    {
+        internal RecommendationExperiment(MLContext context, RecommendationExperimentSettings settings)
+            : base(context,
+                  new RegressionMetricsAgent(context, settings.OptimizingMetric),
+                  new OptimizingMetricInfo(settings.OptimizingMetric),
+                  settings,
+                  TaskKind.Recommendation,
+                  TrainerExtensionUtil.GetTrainerNames(settings.Trainers))
+        {
+        }
+
+        private protected override CrossValidationRunDetail<RegressionMetrics> GetBestCrossValRun(IEnumerable<CrossValidationRunDetail<RegressionMetrics>> results)
```

Comment:
I haven't experimented with SamplingKey and CrossValidation yet. 

cc: @justinormont